### PR TITLE
Feature/messaging

### PR DESCRIPTION
**Pull request by lpradovera**

### CODE REVIEW

**Comment by lpradovera** [comment]
@bryanrite please start taking a look. I could not test reception, everything looks good on the wire but messages don't show up, either in Relay or the SW dashboard.

File: examples/relay/messaging/messaging_receive.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  signalwire
+].each { |f| require f }
+
+# Set logging to debug for testing
+Signalwire::Logger.logger.level = ::Logger::DEBUG
+
+class MessageSendConsumer < Signalwire::Relay::Consumer
+  contexts ['incoming']
+
+  def on_message(message)
```
Comment (bryanrite) [inline examples/relay/messaging/messaging_receive.rb]: This looks incorrect, RCI uses `onIncomingMessage` and `onMessageStateChange`

File: lib/signalwire/relay/messaging.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+      def_delegators :@client, :relay_execute, :protocol, :on, :once, :broadcast
+
+      def initialize(client)
+        @client = client
+        setup_events
+      end
+
+      def send(from_number:, to_number:, context:, **params)
+        body = params.delete(:body)
```
Comment (bryanrite) [inline lib/signalwire/relay/messaging.rb]: https://github.com/signalwire/signalwire-ruby/pull/24/files#diff-e098efa94866bd8afc980b042607487fR21-R28

This section is a bit weird. You remove from params, set payload equal to params, then re-add to payload, then readd payload into params, seems like unnecessary juggling... not a big deal, but looks weird.

I would also remove the error checking. Error checking is done on the server side, don't need to enforce it in the client. We might add a feature like a campaign, or AI integration, that selects body/media from the server side, in which case this validation is invalid.

**Review by bryanrite — CHANGES_REQUESTED** [review]
Couple small changes, but looking good so far!

**Comment by lpradovera** [comment]
@bryanrite ready for review.

**Review by bryanrite — APPROVED** [review]
🚀 🚀

**Pull request merged**